dashboard/test/controllers/activities_controller_test.rb

```diff
@@ -715,42 +624,6 @@ def expect_controller_logs_milestone_regexp(regexp)
     assert_equal_expected_keys build_try_again_response, JSON.parse(@response.body)
   end
 
-  test "anonymous milestone with image saves image but does not save to gallery" do
-    sign_out @user
-
-    # set up existing session
-    client_state.set_level_progress(@script_level_prev.level_id, 50)
-    client_state.add_lines(10)
-
-    # do all the logging
-    @controller.expects :log_milestone
-    @controller.expects :slog
-
-    @controller.expects(:trophy_check).never # no trophy if not logged in
-
-    expect_s3_upload
-
-    assert_creates(LevelSource, LevelSourceImage) do
```

Comment:
Many of the assertions in this test are still relevant; we want to keep this test and only get rid of the assertions about client_state (since that state is now updated on the client.)